pallets/market/src/lib.rs

```diff
@@ -377,6 +399,12 @@ pub mod pallet {
 			let sender = ensure_signed(origin)?;
 
 			let pool_info = Self::get_pool_info(pool_id)?;
+			// check assets id, foundation has no veto
+			ensure!(
+				!T::NontransferableTokens::contains(&pool_info.pool.0) &&
+					!T::NontransferableTokens::contains(&pool_info.pool.1),
+				Error::<T>::NontransferableToken
+			);
 			Self::check_assets_allowed(pool_info.pool)?;
```

Comment:
perhaps the foundation must be able to do this. how do they will mint more liq in the case the pool go bananas?